build.gradle

```diff
@@ -46,12 +43,27 @@ dependencies {
     implementation 'com.google.guava:guava'
     implementation 'com.netflix.nebula:nebula-dependency-recommender'
 
-    testCompile 'com.netflix.nebula:nebula-test'
+    testCompile platform('org.junit:junit-bom')
+    testImplementation 'com.netflix.nebula:nebula-test'
+    testImplementation "org.junit.jupiter:junit-jupiter-api"
+    testRuntimeOnly "org.junit.vintage:junit-vintage-engine"
 
     annotationProcessor "org.immutables:value"
     compileOnly "org.immutables:value::annotations"
 }
 
+tasks.withType(Test) {
+    useJUnitPlatform()
+
+    systemProperties = [
+            'junit.platform.output.capture.stdout'                  : 'true',
+            'junit.platform.output.capture.stderr'                  : 'true',
+            'junit.jupiter.execution.parallel.enabled'              : 'true',
+            'junit.jupiter.execution.parallel.config.strategy'      : 'dynamic',
+            'junit.jupiter.execution.parallel.config.dynamic.factor': '1'
+    ]
+}
```

Comment:
I think I'd like to bake this into baseline before we proliferate it any further.

Reply:
output capture is also pretty annoying if you want to have a quick look at what your test is doing. Likely most cases you don't care but if you do you have to wait until end of the run

Reply:
I've found `--tests` pretty seamless for that?

Perhaps I could say locally do the testLogging but keep it clean on CI?

Reply:
Sure - you could then just log failed? Otherwise you won’t know until ci finishes